core/ipfs/node/node.go

```diff
@@ -0,0 +1,65 @@
+package ipfs
+
+import (
+	"os/exec"
+	"context"
+
+	"github.com/FleekHQ/space-daemon/config"
+	"github.com/FleekHQ/space-daemon/log"
+
+	process "github.com/jbenet/goprocess"
+)
+
+type IpfsNode struct {
+	proc  	  process.Process
+	IsRunning bool
+	Ready     chan bool
+	cfg       config.Config
+}
+
+func NewIpsNode(cfg config.Config) *IpfsNode {
+	return &IpfsNode{
+		Ready: make(chan bool),
+		cfg:   cfg,
+	}
+}
+
+func (node *IpfsNode) Start(ctx context.Context) error {
+    log.Printf("Starting the ipfs node")
+
+    proc := process.WithParent(process.Background())
+
+    proc.Go(func(p process.Process) {
+		cmd := exec.Command("space-ipfs-node")
```

Comment:
Should we handle the scenario where the user is running their own IPFS node? Not sure if this should be a flag when starting space daemon (something like `--use-custom-local-ipfs`). What do you think?

Reply:
This is definitely something to think about.  Should the daemon also detect if IPFS is running by maybe checking the HTTP API port or the `~/.ipfs` directory and make sure we don't overwrite.

Reply:
@dmerrill6 @jsonsivar 

What do you think about reusing `ipfsAddr` for this purpose?

```
core/ipfs/ipfs.go
50:	ipfsAddr := cfg.GetString(config.Ipfsaddr, "/ip4/127.0.0.1/tcp/5001")
52:	multiaddress, err := ma.NewMultiaddr(ipfsAddr)
```

It seems to already default to `/ip4/127.0.0.1/tcp/5001` so the implicit assumption of using the local node is here.

Reply:
Yeah sounds like that's a good idea. So if it was already listening on that port it would give an error right?

Reply:
@jsonsivar 

Yes, but... we can check if `config.Ipfsaddr` equals to `"/ip4/127.0.0.1/tcp/5001"` and then, only then, start our node. What do you think?

Reply:
Ah I see what you mean. Makes sense to me.

Reply:
Not empty though, if its set to localhost we still want to start up the node. Since `core/ipfs/ipfs.go` is the client

Reply:
@jsonsivar 

> Not empty though, if its set to localhost we still want to start up the node.

But what if the user wants to start his own node? 

My reasoning: if the user _explicitly_ asks us to connect to the user node (ie: by setting `config.Ipfsaddr` to `/ip4/127.0.0.1/tcp/5001`) then we should respect that.

Otherwise (ie: if `config.Ipfsaddr` is empty), we should start our own node and set `config.Ipfsaddr` accordingly (ie: `/ip4/127.0.0.1/tcp/5001` by default) since it's used by `core/ipfs/ipfs.go` client, and `space` actually becomes both server and client now.

Any node configuration should use different configuration (`config.IpfsServerAddr`) that would eventually set `config.Ipfsaddr` if blank.

Reply:
That makes a lot of sense :+1: